src/commands/dsyms/api.ts

```diff
@@ -0,0 +1,43 @@
+import {AxiosPromise, AxiosRequestConfig, AxiosResponse} from 'axios'
+import FormData from 'form-data'
+import fs from 'fs'
+import {Writable} from 'stream'
+
+import {buildPath, getRequestBuilder} from '../../helpers/utils'
+
+import {Payload} from './interfaces'
+import {renderUpload} from './renderer'
+import {zip} from './utils'
+
+const maxBodyLength = Infinity
+
+export const uploadDSYM = (request: (args: AxiosRequestConfig) => AxiosPromise<AxiosResponse>) => async (
+  dSYM: Payload,
+  write: Writable['write']
+) => {
+  const form = new FormData()
+  write(renderUpload(dSYM))
+  form.append('type', dSYM.type)
+  const concatUUIDs = dSYM.uuids.join()
+  form.append('uuids', concatUUIDs)
+
+  const tmpFilePath = buildPath('/tmp', 'datadog-ci', 'dsyms', `${concatUUIDs}.zip`)
+  await zip(dSYM.path, tmpFilePath)
```

Comment:
Should we delete the temp directories after use, to avoid the tmp dir to run out of space over time?

Reply:
i switched to [`os.tmpdir()`](https://nodejs.org/api/os.html#os_os_tmpdir), OS should take care of this now

Reply:
It looks safer if the program is handling the clean-up itself rather than relying on the OS to clean up (each OS might clean up at a different rate). It looks like [macOS cleans it up every 3 days](https://superuser.com/questions/187071/in-macos-how-often-is-tmp-deleted).

Since we use  the `tmpFolder` const for the whole execution, maybe we can simply delete that temp file when the command exits?

Reply:
i think having the files for a while can be useful. for example, we can ask the user to look at that folder in case of any problem.
and if the disk space gets too low, this folder should be deleted immediately (that's how it works in iOS).

so i can't think of any problems with using `tmpdir()`, can you?

Reply:
i still think we should let OS deal with this but i've just tried to clear `tmpFolder` at the end of `execute()` as explained [here in Node.js docs](https://nodejs.org/api/fs.html#fs_fspromises_rmdir_path_options)
```typescript
import { promises } from 'fs'
public async execute() {
...
  await promises.rm(tmpFolder, { recursive: true, force: true })
}
```
but compiler gives me the error below
> Property 'rm' does not exist on type 'typeof promises'.ts(2339)

i can only import `rmdir` from `fs` and `fsPromises` packages.
it looks like `rm` was added in v14.14.0, are we using an older version of Node?

Reply:
@lraucy we discussed with @m-rousse over slack and our Node version doesn't have an easy `rm -rf` like functionality.
given that we create temporary files and not in huge volumes, we decided to leave that for later. 
Datadog iOS app's dSYMs are ~25Mb, i imagine the worst case scenario could be ~250Mb.

we will probably use another 3rd party for this (eg: [rimraf](https://www.npmjs.com/package/rimraf))